src/Form/Type/EnableTwoFactorAuthType.php

```diff
@@ -25,13 +27,18 @@ class EnableTwoFactorAuthType extends AbstractType
 {
     public function buildForm(FormBuilderInterface $builder, array $options): void
     {
-        $builder->add('code', TextType::class);
+        $builder->add('code', TextType::class, [
+            'constraints' => [new TwoFactorCodeConstraint($options['user'])]
+        ]);
     }
 
     public function configureOptions(OptionsResolver $resolver): void
     {
-        $resolver->setDefaults([
-            'data_class' => EnableTwoFactorRequest::class,
-        ]);
+        $resolver
+            ->setDefaults([
+                'data_class' => EnableTwoFactorRequest::class,
+                'user' => null,
```

Comment:
adding a `null` default is broken, as `null` is not a valid value (if you don't provide a user, `new TwoFactorCodeConstraint($options['user'])` will produce a TypeError with the null default).
This should define a required option instead of an option with a default value.